src/scaffoldmaker/scaffoldpackage.py

```diff
@@ -253,6 +255,94 @@ def generate(self, region, applyTransformation=True):
             if applyTransformation:
                 self.applyTransformation()
 
+    def deleteElementsInRanges(self, region, deleteElementRanges):
+        """
+        If this is the root scaffold and there are ranges of element identifiers to delete,
+        remove these from the model.
+        Also check if marker group nodes embedded in those elements are shared with nearby elements and delete those
+        marker nodes if used only by the deleted elements.
+        :param deleteElementRanges: Range of elements to be deleted.
+        """
+        if (len(deleteElementRanges) == 0):
+            return
+
+        self._region = region
+        fm = self._region.getFieldmodule()
+        mesh = get_highest_dimension_mesh(fm)
+        meshDimension = mesh.getDimension()
+        nodes = fm.findNodesetByFieldDomainType(Field.DOMAIN_TYPE_NODES)
+        with ChangeManager(fm):
+            # put the elements in a group and use subelement handling to get nodes in use by it
+            destroyGroup = fm.createFieldGroup()
+            destroyGroup.setSubelementHandlingMode(FieldGroup.SUBELEMENT_HANDLING_MODE_FULL)
+            destroyElementGroup = destroyGroup.createFieldElementGroup(mesh)
+            destroyMesh = destroyElementGroup.getMeshGroup()
+            elementIter = mesh.createElementiterator()
+            element = elementIter.next()
+            while element.isValid():
+                identifier = element.getIdentifier()
+                for deleteElementRange in deleteElementRanges:
+                    if (identifier >= deleteElementRange[0]) and (identifier <= deleteElementRange[1]):
+                        destroyMesh.addElement(element)
+                element = elementIter.next()
+            del elementIter
+            # print("Deleting", destroyMesh.getSize(), "element(s)")
+            if destroyMesh.getSize() > 0:
+                fieldcache = fm.createFieldcache()
+                destroyNodeGroup = destroyGroup.getFieldNodeGroup(nodes)
+                destroyNodes = destroyNodeGroup.getNodesetGroup()
+                markerGroup = fm.findFieldByName("marker").castGroup()
+                if markerGroup.isValid():
+                    markerNodes = markerGroup.getFieldNodeGroup(nodes).getNodesetGroup()
+                    markerLocation = fm.findFieldByName("marker_location")
```

Comment:
Import/use utility function from annotationgroup.py (put in to avoid field names being spread around the codee and potentially leading to typo errors):
```
markerLocation = getAnnotationMarkerLocationField(fieldmodule, mesh)
```
Remove next line for markerName (which if needed should use similar getAnnotationMarkerNameField()).